packages/next-on-pages/templates/_worker.js/utils/cache.ts

```diff
@@ -55,7 +55,12 @@ export async function handleSuspenseCacheRequest(request: Request) {
 				// Update the value in the cache.
 				const body = await request.json<IncrementalCacheValue>();
 				// Falling back to the cache tags header for Next.js 13.5+
-				body.tags = body.tags ?? body.data.tags ?? request.headers.get(CACHE_TAGS_HEADER)?.split(',') ?? [];
+				if (body.data.tags === undefined) {
+					body.tags = body.tags ?? request.headers.get(CACHE_TAGS_HEADER)?.split(',')?.filter(Boolean) ?? [];
+				} else {
+					body.tags = body.data.tags;
+					delete body.data.tags;
```

Comment:
Not sure whether to have this or not - because we're also checking for `body.data.tags` in retrieval phase

Reply:
Getting from the cache checks `body.data.tags` for backwards compatibility. Deleting here is not going to impact that since this it also checks `body.tags` which is what this will use instead.

Reply:
Is deleting the `tags` field here necessary/beneficial in any way? if not I would suggest not to do it

Reply:
> Is deleting the `tags` field here necessary/beneficial in any way? if not I would suggest not to do it

storing less data in the cache if the user is on an older version of next.js. no need to store the tags twice. while it would be a small amount of extra data, things like kv still bill for data stored.

Reply:
yeah, sorry, my bad, you're right :+1: